nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -259,6 +262,18 @@ Costmap2DROS::on_configure(const rclcpp_lifecycle::State & /*state*/)
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
+void Costmap2DROS::getCostCallback(
+  const std::shared_ptr<rmw_request_id_t> /*request_header*/,
+  const std::shared_ptr<nav2_msgs::srv::GetCost::Request> request,
+  const std::shared_ptr<nav2_msgs::srv::GetCost::Response> response)
+{
+  RCLCPP_INFO(
+    get_logger(), "Received request to get cost at point (%f, %f)", request->x, request->y);
+  auto cost = static_cast<float>(layered_costmap_->getCostmap()->getCost(
```

Comment:
I think some bounds checking needs to happen that the request is within the costmap's size